core/trino-main/src/main/java/io/trino/cost/FilterStatsCalculator.java

```diff
@@ -105,6 +114,31 @@ public PlanNodeStatsEstimate filterStats(
                 .process(simplifiedExpression);
     }
 
+    @VisibleForTesting
+    Optional<Symbol> getSymbolReference(Expression expression, Session session, TypeProvider types)
+    {
+        if (expression instanceof SymbolReference) {
+            return Optional.of(Symbol.from(expression));
+        }
+        if (!(expression instanceof Cast)) {
+            return Optional.empty();
+        }
+        Cast castExpression = (Cast) expression;
+        if (!(castExpression.getExpression() instanceof SymbolReference)) {
+            return Optional.empty();
+        }
+        Map<NodeRef<Expression>, Type> expressionTypes = typeAnalyzer.getTypes(session, types, expression);
+        Type castSourceType = expressionTypes.get(NodeRef.of(castExpression.getExpression()));
+        Type castTargetType = expressionTypes.get(NodeRef.<Expression>of(castExpression));
+        // CAST must be an implicit coercion and the stats representation of target type
```

Comment:
> must be an implicit coercion

explain why.
we expect implicit coercion 

- to be generally injective... which is not true for coercion to double, so maybe worth also documenting why it is not a problem here. Compare with `io.trino.sql.planner.iterative.rule.UnwrapCastInComparison.Visitor#hasInjectiveImplicitCoercion` logic. I don't think we need to be as complex (this is stats estimates only), but we should document assumptions made
- to be order-preserving (see assumption documentation in `UnwrapCastInComparison` about "order-preserving")